packages/flutter/lib/src/rendering/custom_paint.dart

```diff
@@ -817,6 +817,21 @@ class RenderCustomPaint extends RenderProxyBox {
     if (properties.button != null) {
       config.isButton = properties.button;
     }
+    if (properties.textField != null) {
+      config.isTextField = properties.textField;
+    }
+    if (properties.focused != null) {
+      config.isFocused = properties.focused;
```

Comment:
I'm worried that we might be conflating "focused" with "currently being edited", which on touch screens are two different things 😕 But I think short of a big API change, this ship might have sailed.

Reply:
How are they different on touch screens? We do make a pretty strong difference between input focus (which is what this is) and accessibility focus.

Reply:
On non-touch screens a text field would enter focused state immediately, while on touch screens it needs to be activated. But yeah, if focus and accessibility focus are separated, then we're good.